### PR TITLE
add a cmake config

### PR DESCRIPTION
**Pull request by juliendehos**
Thank you for developing cubature.
Here is a cmake config which enables to integrate cubature easily in a cmake project (for example, https://github.com/juliendehos/cubature_test).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+project( cubature )
+
+set( CMAKE_C_FLAGS " -fPIC ${CMAKE_C_FLAGS}" )
+
+find_package( PkgConfig REQUIRED )
+pkg_check_modules( PKG_FFTW REQUIRED fftw3l )
+include_directories( ${PKG_FFTW_INCLUDE_DIRS} )
+
+add_library( cubature SHARED 
+    clencurt_gen.c
```
Comment (stevengj) [inline CMakeLists.txt]: This file is only used to generate clencurt.h, and doesn't need to be linked into the library.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required( VERSION 3.0 )
+
+project( cubature )
+
+set( CMAKE_C_FLAGS " -fPIC ${CMAKE_C_FLAGS}" )
+
+find_package( PkgConfig REQUIRED )
+pkg_check_modules( PKG_FFTW REQUIRED fftw3l )
```
Comment (stevengj) [inline CMakeLists.txt]: FFTW isn't required, because the generated file `clencurt.h` is included in the git repo.

**Comment by juliendehos** [comment]
Yes. Fixed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required( VERSION 3.0 )
+
+project( cubature )
+
+set( CMAKE_C_FLAGS " -fPIC ${CMAKE_C_FLAGS}" )
```
Comment (stevengj) [inline CMakeLists.txt]: `-fPIC` should be set by default when building a shared library, no? https://cmake.org/cmake/help/v3.4/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE

Comment (juliendehos) [reply CMakeLists.txt]: Yes, fixed.
I didn't know this, thank you.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+add_library( cubature SHARED 
+    hcubature.c
+    pcubature.c)
+
+add_executable( htest test.c )
+target_link_libraries( htest cubature m )
+
+add_executable( ptest test.c )
+target_link_libraries( ptest cubature m )
+set_target_properties( ptest PROPERTIES COMPILE_FLAGS "-DPCUBATURE" )
```
Comment (stevengj) [inline CMakeLists.txt]: Shouldn't this be https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html ?

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+install( TARGETS cubature DESTINATION lib )
```
Comment (stevengj) [inline CMakeLists.txt]: Should be CMAKE_INSTALL_LIBDIR and CMAKE_INSTALL_INCLUDEDIR

Comment (juliendehos) [reply CMakeLists.txt]: Ok, fixed (target property and install dirs).

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+add_library( cubature SHARED 
+    hcubature.c
+    pcubature.c)
+
+add_executable( htest test.c )
+target_link_libraries( htest cubature m )
+
+add_executable( ptest test.c )
+target_link_libraries( ptest cubature m )
+target_compile_definitions( ptest PRIVATE "-DPCUBATURE" )
```
Comment (stevengj) [inline CMakeLists.txt]: I thought the syntax was something like
```
target_compile_definitions( ptest PRIVATE PCUBATURE=1 )
```

Comment (juliendehos) [reply CMakeLists.txt]: Ok. Both versions seem to work.

Comment (stevengj) [reply CMakeLists.txt]: The advantage of `PCUBATURE=1` is that `cmake` will generate the appropriate compiler flag on non-Unixy compilers.

Comment (juliendehos) [reply CMakeLists.txt]: Ok. Thanks for this information.

**Pull request merged**